sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1396,6 +1398,243 @@ def pct_change_overload(df, periods=1, fill_method='pad', limit=None, freq=None)
+        func_lines += [f'  return pandas.DataFrame({{{data}}}, index={df}._index)']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'numpy': numpy,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def sdc_pandas_dataframe_isin_dict(df, values, func_name):
+    all_params = ['df', 'values']
+
+    df_func_name = f'_df_{func_name}_impl'
+    func_text, global_vars = sdc_pandas_dataframe_isin_dict_codegen(func_name, df, values, all_params, df.columns)
+
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _apply_impl = loc_vars[df_func_name]
+    return _apply_impl
+
+
+def sdc_pandas_dataframe_isin_ser_codegen(func_name, df_type, values, all_params, columns):
+    result_name = []
+    joined = ', '.join(all_params)
+    func_lines = [f'def _df_{func_name}_impl({joined}):']
+    df = all_params[0]
+    for i, c in enumerate(columns):
+        result_c = f'result_{c}'
+        func_lines += [
+            f'  series_{c} = pandas.Series(get_dataframe_data({df}, {i}))',
+            f'  result = numpy.empty(len(series_{c}._data), numpy.bool_)'
+        ]
+        if isinstance(values.index, types.NoneType):
+            func_lines += [
+                f'  for i in range(len(series_{c}._data)):',
+                f'    if series_{c}._data[i] == values._data[i]:',
+                f'      result[i] = True',
+                f'    else:',
+                f'      result[i] = False',
+            ]
+        else:
+            func_lines += [
+                f'  for i in range(len(series_{c})):',
+                f'    idx = {df}._index[i]',
+                f'    value = series_{c}._data[i]',
+                f'    for j in numba.prange(len(values)):',
+                f'      idx_val = values._index[j]',
+                f'      if idx == idx_val:',
+                f'        value_val = values._data[j]',
+                f'        if value == value_val:',
+                f'          result[i] = True',
+                f'          break',
+                f'        else:',
+                f'          result[i] = False',
+                f'      else:',
+                f'        result[i] = False'
+            ]
+
+        func_lines += [f'  {result_c} = pandas.Series(result)']
+        result_name.append((result_c, c))
+
+    data = ', '.join(f'"{column_name}": {column}' for column, column_name in result_name)
+    if isinstance(df_type.index, types.NoneType):
+        func_lines += [f'  return pandas.DataFrame({{{data}}})']
+    else:
+        func_lines += [f'  return pandas.DataFrame({{{data}}}, index={df}._index)']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'numba': numba,
+                   'numpy': numpy,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def sdc_pandas_dataframe_isin_ser(df, values, func_name):
```

Comment:
The functions:
    sdc_pandas_dataframe_isin_dict
    sdc_pandas_dataframe_isin_ser
    sdc_pandas_dataframe_isin_df
look almost identical and differ only in the applied function code generator.
I think you can apply the same approach as was used for getitem, see: [hpat_pandas_dataframe_functions.py](https://github.com/IntelPython/sdc/blob/ec7e4f7fd3325391285b4856c092fb24b957e36f/sdc/datatypes/hpat_pandas_dataframe_functions.py#L1367)